### DISCUSSION TITLE
What is the best way to use $disconnect()

### DISCUSSION THREAD

**Discussion by indahud** [prisma/prisma#5890] [Q&A] [answered]
I am using NextJS and Prisma for my project. 
I have different pages inside /pages and I am using prisma to fetch data from database.
I have created seprate /lib/prisma.js file to connect and disconnect prisma.

```JavaScript
// prisma.js file in /lib

import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

/* 
 * Try 1 function to disconnect prisma
*/
async function prismaDisconnect() {
  try {
    const prismaDisconnect = await prisma.$disconnect();
    // console.log("Disconnected");
  } catch (error) {
    console.error(error);
  }
}

/* Try 2 
* Cretad empty function and invoked it like documentation
*/
async function main() {}

main()
  .catch((e) => {
    console.log(e);
    process.exit(1);
  })
  .finally(() => {
    prisma.$disconnect();
  });

export { prisma, main, prismaDisconnect };

```
```JavaScript
// Items.js in /pages

import { prisma, prismaDisconnect, main } from "../lib/prisma"

export const getStaticProps = async () => {
    try {
        const items = await prisma.items.findMany();
        return {
        props: {
            items,
        },
    };
  } catch (error) {
    console.error(error);
  } finally {
    // try 1 Directly calling not working  
    const prismaDisconnect = await prisma.$disconnect();
    // try 2 Created function in ../lib invoked here still not working
    prismaDisconnect()
    // invoked main function defined in /lib
    // Working but command line warning 
    main
  }
}
```
For demonstration puropse I shared the code above. Calling prismaDisconnect throws an error 

`Unhandled Runtime Error. Error: PrismaClient is unable to be run in the browser` 

Invoking the main function does not throw any error but it seems to be not working becuase there is a warning message in command line `warn(prisma-client) Already 10 Prisma Clients are actively running.`

Everything is working smoothly if I dont use `prismadisconnect`

Really grateful for any help

**Comment by ryands17** [comment, marked as answer]
@indahud 👋 

You do not need to use `$disconnect` in your Next.js app. We have specified [here](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management#connection-management-in-serverless-environments) that in serverless environments it's not recommended to call `$disconnect` explicitly.

As for using Prisma with Next.js, you can follow [this](https://www.prisma.io/nextjs) guide where we have explained how to use Prisma inside Next.js with examples on GitHub :)

Let me know if you have any further queries.

**Reply by indahud** [reply]
@ryands17 Thank you. There is a still warning on command line `warn(prisma-client) Already 10 Prisma Clients are actively running`. I am working on small websitsite which has 10-12 pages. I am visiting all those page links for testing. Everything is working fine except that warning. I also added connection limit.

**Reply by ryands17** [reply]
@indahud For this you can follow [this](https://www.prisma.io/docs/support/help-articles/nextjs-prisma-client-dev-practices) post in the docs where you need to instantiate `PrismaClient` in a special way for development.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
